### DISCUSSION TITLE
Is there a way to support an additional redirect on the service callback redirect?

### DISCUSSION THREAD

**Discussion by fubuloubu** [fief-dev/fief#405] [Q&A] [answered]
I was trying this, but any additional query params on the configured callback uri fails the check within fief that the callback is as expected for the given client

Basically, when Fief handles my service's callback, either want it to ignore or specifically whitelist having `redirect_uri=/some/other/path?maybe=some-params&...` (encoded ofc) in the callback's params, so when I receive the auth code via the callback on my service, I can create another redirect back to where the user came from that triggered the auth redirect sequence originally in my service

I need this to handle invites within my service for new users

This is how I have my client configured currently:
![image](https://github.com/user-attachments/assets/a5d839d1-8d5e-4472-a230-fd93215ced4e)

**Comment by frankie567** [comment, marked as answer]
For this, I would recommend you to use the standard OAuth2 [`state` parameter](https://docs.fief.dev/going-further/authorize-url/). The value can be anything you want: a JWT, a key referring to data stored in a DB or key-value store, etc.

**Reply by fubuloubu** [reply]
Interesting! Now that is some good advice!

**Reply by fubuloubu** [reply]
Ended up doing this, just generate a URL-safe key to use for `state` and set the value to some data structure including redirect info and such. If you intend on using this in a multi-service architecture, make sure you cache it appropiately using Redis or something (just good practice anyways)
